manifests/repo/yum.pp

```diff
@@ -4,73 +4,13 @@
-        enabled  => '1',
-        gpgcheck => '1',
-        gpgkey   => 'https://yum.puppetlabs.com/RPM-GPG-KEY-puppet',
-      }
-    } else {
-
-      yumrepo { 'puppetlabs-products':
-        descr    => "Puppet Labs Products EL ${::operatingsystemmajrelease} - ${::architecture}",
-        baseurl  => "http://yum.puppetlabs.com/el/${::operatingsystemmajrelease}/products/${::architecture}",
-        enabled  => '1',
-        gpgcheck => '1',
-        gpgkey   => 'https://yum.puppetlabs.com/RPM-GPG-KEY-puppet',
-      }
-      yumrepo { 'puppetlabs-products-source':
-        descr    => "Puppet Labs Products EL ${::operatingsystemmajrelease} - ${::architecture} - Source",
-        baseurl  => "http://yum.puppetlabs.com/el/${::operatingsystemmajrelease}/products/SRPMS",
-        enabled  => $source_enable,
-        gpgcheck => '1',
-        gpgkey   => 'https://yum.puppetlabs.com/RPM-GPG-KEY-puppet',
-      }
-
-      yumrepo { 'puppetlabs-deps':
-        descr    => "Puppet Labs Dependencies EL ${::operatingsystemmajrelease} - ${::architecture}",
-        baseurl  => "http://yum.puppetlabs.com/el/${::operatingsystemmajrelease}/dependencies/${::architecture}",
-        enabled  => '1',
-        gpgcheck => '1',
-        gpgkey   => 'https://yum.puppetlabs.com/RPM-GPG-KEY-puppet',
-      }
-      yumrepo { 'puppetlabs-deps-source':
-        descr          => "Puppet Labs Dependencies EL ${::operatingsystemmajrelease} - ${::architecture} - Source",
-        baseurl        => "http://yum.puppetlabs.com/el/${::operatingsystemmajrelease}/dependencies/SRPMS",
-        enabled        => $source_enable,
-        gpgcheck       => '1',
-        gpgkey         => 'https://yum.puppetlabs.com/RPM-GPG-KEY-puppet',
-        failovermethod => 'priority'
-      }
-
-      yumrepo { 'puppetlabs-devel':
-        descr    => "Puppet Labs Devel EL ${::operatingsystemmajrelease} - ${::architecture}",
-        baseurl  => "http://yum.puppetlabs.com/el/${::operatingsystemmajrelease}/devel/${::architecture}",
-        enabled  => $devel_enabled,
-        gpgcheck => '1',
-        gpgkey   => 'https://yum.puppetlabs.com/RPM-GPG-KEY-puppet',
-      }
-      yumrepo { 'puppetlabs-devel-source':
-        descr    => "Puppet Labs Devel EL ${::operatingsystemmajrelease} - ${::architecture} - Source",
-        baseurl  => "http://yum.puppetlabs.com/el/${::operatingsystemmajrelease}/devel/SRPMS",
-        enabled  => $source_enable,
-        gpgcheck => '1',
-        gpgkey   => 'https://yum.puppetlabs.com/RPM-GPG-KEY-puppet',
-      }
+    $lc_collection_name = downcase($puppet::collection)
+    yumrepo { "puppetlabs-${lc_collection_name}":
+      descr    => "Puppet Labs Collection ${::puppet::collection} EL ${::operatingsystemmajrelease} - ${::architecture}",
+      baseurl  => "http://yum.puppetlabs.com/el/${::operatingsystemmajrelease}/${::puppet::collection}/${::architecture}",
```

Comment:
As far as I know, Puppet 5 packages are available in a separate repository for rpm-based distros, e.g. http://yum.puppetlabs.com/puppet5/el/7/x86_64 (note that puppet5 goes before /el/ part), as it's mentioned in [puppet 5 installing instructions](https://puppet.com/docs/puppet/5.5/puppet_platform.html#using-puppet-5-platform). Not sure if it's directly connected with this PR though.

Reply:
Thanks.
I hadn't tested this on RedHatish distros or dug too far into the docs.